src/ConcreteOperations/volume.jl

```diff
@@ -0,0 +1,34 @@
+export volume
+
+"""
+    volume(H::AbstractHyperrectangle)
+
+Return the volume of a hyperrectangular set.
+
+### Input
+
+- `H` -- hyperrectangular set
+
+### Output
+
+The (Euclidean) volume of ``H``.
+
+### Algorithm
+
+The volume of the hyperrectangle ``H`` with vector radius ``r`` is
+``2ⁿ ∏ᵢ rᵢ`` where ``rᵢ`` denotes the ``i``-th component of ``r``.
+"""
+function volume(H::AbstractHyperrectangle{N}) where {N<:AbstractFloat}
+    r = radius_hyperrectangle(H)
+    n = dim(H)
+    α = exp(n*log(2))
+    vol = α * prod(r)
+    return vol
+end
```

Comment:
This implementation is faster than your fallback but slightly less precise and slower than my second suggestion in the next comment (but I copied the functions here for convenience).
```julia
function volume2(H::AbstractHyperrectangle{N}) where {N<:Real}
    vol = one(N)
    for ri in radius_hyperrectangle(H)
        vol *= 2 * ri
    end
    return vol
end

function volume3(H::AbstractHyperrectangle{N}) where {N<:Real}
    vol = one(N)
    for i in 1:dim(H)
        vol *= 2 * radius_hyperrectangle(H, i)
    end
    return vol
end

julia> B = BallInf(zeros(100), 0.5);

julia> @btime volume($B)
  120.164 ns (1 allocation: 896 bytes)
1.0000000000000018

julia> @btime volume2($B)
  153.538 ns (1 allocation: 896 bytes)
1.0

julia> @btime volume3($B)
  75.993 ns (0 allocations: 0 bytes)
1.0
```